src/array/specification.rs

```diff
@@ -79,7 +79,8 @@ pub fn check_offsets_and_utf8<O: Offset>(offsets: &[O], values: &[u8]) -> usize
         let end = window[1].to_usize();
         assert!(end <= values.len());
         let slice = unsafe { std::slice::from_raw_parts(values.as_ptr().add(start), end - start) };
-        std::str::from_utf8(slice).expect("A non-utf8 string was passed.");
+        #[cfg(not(feature = "simdutf8"))]
```

Comment:
I think this introduced a vulnerability as it no longer checks utf8 if `simdutf8` is disabled. Both branches should be here.

Reply:
Ah - I meant to remove this `cfg` here as it's no longer optional.

Reply:
you are right since this is not optional (also think thinking the previous xD). Pushed a fix directly to main